Customer Home: Add Mastering Gutenberg Card

Aurorum opened the pull request:
#### Changes proposed in this Pull Request

This is very much a WIP; I still need to create the videos with some more detailed annotations/text, and select a thumbnail for them, along with making sure the support links are localised etc.

This adds another card with some Gutenberg resources.

#### Testing instructions

Copy and paste this into `client/my-sites/customer-home/locations/primary/index.jsx`

```jsx
/**
 * External dependencies
 */
import React from 'react';

/**
 * Internal dependencies
 */
import GoMobile from 'my-sites/customer-home/cards/features/go-mobile';
import ChecklistSiteSetup from 'my-sites/customer-home/cards/primary/checklist-site-setup';
import MasteringGutenberg from 'my-sites/customer-home/cards/education/mastering-gutenberg';
import QuickLinks from 'my-sites/customer-home/cards/primary/quick-links';

const cardComponents = {
	'home-feature-go-mobile-phones': GoMobile,
	'home-primary-checklist-site-setup': ChecklistSiteSetup,
	'home-primary-quick-links': QuickLinks,
	'home-education-mastering-gutenberg': MasteringGutenberg,
};

const Primary = ( { checklistMode, cards } ) => {
	// Always ensure we have primary content.
	if ( cards && cards.length < 1 ) {
		cards = [ 'home-primary-quick-links' ];
	}
	return (
		<div className="primary">
			{ cards &&
				cards.map(
					( card, index ) =>
						cardComponents[ card ] &&
						React.createElement( cardComponents[ card ], {
							key: index,
							checklistMode: card === 'home-primary-checklist-site-setup' ? checklistMode : null,
						} )
				) }
			<MasteringGutenberg />
		</div>
	);
};

export default Primary;

```

A list of links should appear. The card should not appear for sites unable to use Gutenberg.

<img width="722" alt="Screenshot 2020-04-07 at 22 31 11" src="https://user-images.githubusercontent.com/43215253/78721460-81644280-791f-11ea-9e82-0d510b209245.png">

Clicking one should open a modal with a video and a link to a support document on it. Verify that both of these work (the video should autoplay).

<img width="703" alt="Screenshot 2020-04-07 at 22 30 28" src="https://user-images.githubusercontent.com/43215253/78721435-78737100-791f-11ea-8d0a-7fbc767a0687.png">

The card should be dismissable, but hidden if a user cannot use Gutenberg (based on the `isClassicEditorForced` selector). 

Fixes #39132

Aurorum commented:
Just a couple quick questions (cc @danhauk, @gwwar)

1. Are videos like these okay? https://cdn.discordapp.com/attachments/449249666798714890/680849856897745121/adjusting-blocks-tutorial.mp4 (note: auto-downloads)

2. Now that a few more videos are going to be used, is it worth attempting to add them to GitHub through the method mentioned by @mmtr [here](https://github.com/Automattic/wp-calypso/pull/39506#issuecomment-588188073), or would it be better to upload it somewhere else and embed it?

3. With annotations, the videos probably would cause more i18n issues. Would it be worth making the links for non-English users direct straight to the support document, rather than displaying a video that could become quite tricky to understand?

4. When should the card be hidden? The tutorials are fairly basic, so would it make sense to hide for users that have been registered longer? Or would a dismiss option work better? Currently, I've made it so the card is only hidden if the user has opted out of Gutenberg and decided to stick with the Classic Editor

5. The initial issue favoured having the card on the right column, but should this still be the case? I feel like there's a lot more space to be filled on the left column now, and it could work better there.

<img width="1271" alt="Screenshot 2020-02-22 at 17 23 16" src="https://user-images.githubusercontent.com/43215253/75097733-1c8a9e00-55a6-11ea-8028-838b3da0a1c1.png">

simison commented:
Should "Block Editor" be lower case?

These graphics from core iteration on welcome modal might be useful: https://github.com/WordPress/gutenberg/issues/19209#issuecomment-568556537

Aurorum commented:
Both the Welcome Guide in Gutenberg and the initial design don't have it in lower case, so I'm guessing not. :)

<img width="289" alt="Screenshot 2020-02-22 at 22 18 35" src="https://user-images.githubusercontent.com/43215253/75100094-4ac9a700-55c1-11ea-8c7a-91c521785ce9.png">

Aurorum commented:
I'm going to mark this for review now even though it's not technically finished, but I can't really proceed much further without some of the advice with the questions above. :) 

__Proposed Tutorials__

Adding, moving and deleting blocks: https://cdn.discordapp.com/attachments/449249666798714890/680849856897745121/adjusting-blocks-tutorial.mp4

Quickly navigating between blocks: https://cdn.discordapp.com/attachments/449249666798714890/681109889686175746/block-navigation-tutorial.mp4

Adding links: https://cdn.discordapp.com/attachments/449249666798714890/681109899580669974/linkstutorial.mp4

Using images: https://cdn.discordapp.com/attachments/449249666798714890/681115972975067179/using-images-tutorial.mp4

Let me know if these videos are okay. The topics seem to cover the main issues in the forums: https://en.forums.wordpress.com/topic-tag/block-editor/

kwight commented:
This is looking great @Aurorum <3 I'll leave design and such to others, but a couple of quick notes.

* The card seems to have a blue top border not present anywhere else.
* I'm wondering if the link styles can align better with existing cards, one way or the other. Maybe these could drop the icon and add the external icon indicator to the right, more like the card below it:

<img width="476" alt="Screen Shot 2020-02-24 at 10 32 06 AM" src="https://user-images.githubusercontent.com/349751/75180612-6a5fed00-56f1-11ea-9463-3151ea3b6520.png">

Side note that our designers are at a meetup this week, so review will be delayed for a while. Thanks for your patience, we appreciate the contributions! 👍

kwight commented:
> Maybe these could drop the icon and add the external icon indicator to the right

Oh, ignore me – they open modals 🙃

kwight commented on client/my-sites/customer-home/mastering-gutenberg-card/index.jsx:
```diff
@@ -0,0 +1,160 @@
+		} );
+	};
+
+	handleAdjustingBlocksClick = () => {
```
I'm not sure, but we may be able to simplify these methods to just one, if we make state something like

```js
{ showDialog: true,  activeDialog: 'adjustingBlocks' }
```

This would also mean the `if else if` blocks at L60 could be a simpler [switch](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch) statement checking on `activeDialog`.

Aurorum replied:
That works well, thanks! :)

Aurorum replied:
That works well, thanks! :)

dezzie commented:
Hi there @Aurorum, Customer Home designer here, thank you so much for all of your contributions and enthusiasm for this 😍. I don't happen to be at the same meetup as the rest of the design team, and I didn't want to leave you hanging any longer on design feedback and the questions you asked: 

**Overall:**
Aim to keep the content and presentation as consistent as possible, so it doesn't produce a disjointed experience for users. Consistency also helps us narrow down the reasons for why things perform down the line.

**Styling** 
- Can you please remove the blue top border for consistency? Clarity is more important than pretty :). That treatment was more of a design exploration that didn't stick because it wasn't consistent with the rest of the product. We'll try to be better about communicating if/when we've chosen a visual direction and discarded others.

- Are you combining the Close and Learn more buttons? They should be separate. The external link icon on the Learn more button is perfect usage of that type of button.

- We should remove the icons to the left of the 4 links in that card. Because those icons are for affordances (meaning they represent specific actions or functionality in the UI), they shouldn't be used as bullet points or other styling that starts to give them multiple meanings. This messes with users who are trying to learn something new. The noble mission of your work here!

- Don't capitalize "Block Editor," as it isn't the Official Name of the Thing. Gutenberg! Gutenberg!

**Behavior**
- You're smart to point out the different conditions under which the card would be helpful to show or hide. While I agree that it would make sense to hide the card if a person explicitly doesn't want to use Gutenberg, the community keeps adding new functionality and fixing things that didn't work well in a previous release. A Dismiss option works best for serving all of those intentions.

**Layout**
- You're right, everywhere there are mockups of this being on the right-hand side, but as you can see, it's gotten a little cozy :). I think your wider-width design is a better solution :)

mmtr commented:
> Now that a few more videos are going to be used, is it worth attempting to add them to GitHub through the method mentioned by @mmtr [here](https://github.com/Automattic/wp-calypso/pull/39506#issuecomment-588188073), or would it be better to upload it somewhere else and embed it?

Seems it makes sense to start uploading videos directly in Calypso. I'll see if I can open a PR tomorrow adding a video loader to webpack but anyone is welcome to beat me to it (cc @sgomes).

Aurorum commented:
I started one in #39707. :)

Aurorum commented:
Thank you very much @dezzie! I've removed the borders and the icons, and stopped using the `ButtonGroup` component for the two buttons. I've also moved the card to the left column, and made it a dismissable one. It will now always display unless the user cannot use Gutenberg or has dismissed it. 

<img width="731" alt="Screenshot 2020-02-26 at 17 05 17" src="https://user-images.githubusercontent.com/43215253/75368805-2e2bb880-58ba-11ea-9177-f8209dd48c1d.png">
<img width="263" alt="Screenshot 2020-02-26 at 17 03 05" src="https://user-images.githubusercontent.com/43215253/75368574-e147e200-58b9-11ea-96e1-7a5af93da3a8.png">

> Don't capitalize "Block Editor," as it isn't the Official Name of the Thing. Gutenberg! Gutenberg!

I've done so, but I was wondering if this was more inconsistent? Not just because "Block Editor" is used throughout, but more because all of the Customer Home cards have capitalised titles.

dezzie commented:
@Aurorum, you're correct in pointing out the inconsistency with the capitalization. In the future, we'll try to be mindful of things that community contributors might take literally. I'll do my best to update any mockups on here as I come across them. Thank you for being cool with it, keep raising these things up!

Aurorum commented:
Thanks @dezzie! I added a commit to capitalise the title again. :)

kwight commented on client/my-sites/customer-home/mastering-gutenberg-card/index.jsx:
```diff
@@ -0,0 +1,159 @@
+
+class MasteringGutenbergCard extends Component {
+	state = {
+		showDialog: false,
```
Let's update this with the new `activeDialog: false`.

kwight commented on client/my-sites/customer-home/mastering-gutenberg-card/index.jsx:
```diff
@@ -0,0 +1,159 @@
+	handleDialogClosure = () => {
+		this.setState( {
+			showDialog: false,
+			adjustingBlocks: false,
```
These can now all be replaced with `activeDialog: false`.

kwight commented on client/my-sites/customer-home/mastering-gutenberg-card/index.jsx:
```diff
@@ -0,0 +1,159 @@
+		} );
+	};
+
+	handleAdjustingBlocksClick = () => {
```
I think we can further take this to just one handler instead of four, since the only difference between them is a string identifier for the button clicked (which we can [pass as an argument](https://reactjs.org/docs/handling-events.html)). So something like:

```js
handleButtonClick = ( button ) => {
    this.setState( { showDialog: true, activeDialog: button } );
}
```

Then each button would have:

```js
onClick={ this.handleButtonClick.bind( this, 'addingLinks' ) }
```

Aurorum replied:
Yes, that works perfectly and is much cleaner, thank you!

Aurorum commented:
I think the only thing left to do now is just add the videos with their thumbnails. Once #39707 is merged, I'll try and do so, but I just wanted to verify if the videos I linked are okay to use? :)

dezzie commented:
> I think the only thing left to do now is just add the videos with their thumbnails. Once #39707 is merged, I'll try and do so, but I just wanted to verify if the videos I linked are okay to use? :)

I will defer to @danhauk or @gwwar or @lcollette on this one, since they've followed this idea more in depth than I have.

gwwar commented:
Sorry for the delay @Aurorum we're still chatting through where we'd like to host videos for WordPress.com. We'll get back to you soon, I think we're leaning toward hosting these on VideoPress. (We'll be able to help upload related files).

cc @apeatling in case you had any feedback for the videos here: 

https://github.com/Automattic/wp-calypso/pull/39617#issuecomment-590064551

The annotations are a nice touch (though we'll need to circle back to thinking through i18n)! I do think it's likely that we need to bump up the video quality. 

In addition to VideoPress YouTube is also a consideration, so maybe we could have the annotations as subtitles captions?

Aurorum commented:
Thanks @gwwar! For what it's worth, I personally reckon that the annotations as captions is probably the best solution. :)

apeatling commented:
Could we do some user tests on the videos to get a better idea of how they are working? I really like the captions, although I suspect they are still too fast.

Aurorum commented:
> Could we do some user tests on the videos to get a better idea of how they are working/if people can follow them?

I'm not exactly sure how this will be possible to get working with an mp4 video, but if they're being uploaded to YouTube, that'll probably make some statistics available. :)

kwight commented:
Hi @Aurorum ! Thanks again for all the work you're contributing here <3

Just letting you know we're prioritizing some video production internally, so we're a little on hold for now with PRs like this – we'll want to see what comes out of the new videos (which will be on the Automattic YouTube channel) to determine which cards and such are needed to support them. Let's pause here, and we'll circle back when we have more concrete needs for cards. 👍

Aurorum commented:
Thanks for letting me know! 

I'll resolve conflicts and all when/if it's time to proceed. :)

gwwar commented:
Videos are now available in https://www.youtube.com/playlist?list=PLQFhxUeNFfdLM_69zeGats_t_ounaEMuz

@Aurorum if you're interested in picking this one up again, as a heads up we've just completed a number of refactors to the home section, moving logic from the client to the layout API. We should still be able to prototype the card out in devdocs or as a temporary commit to the section in the meantime 👍

Aurorum commented:
Happy to start this one again! The videos seem to include all four aspects of the guide, but they occur at different times. This shouldn't usually be a problem, but it seems that the YouTube `end` parameter isn't working with embeds (although `start` is fine).

Just to clarify, should the list still be those four things or should it be "Adjusting settings for blocks" and "Customizing posts and pages with blocks", in which case the whole video could be played?

<img width="353" alt="Screenshot 2020-04-07 at 09 45 15" src="https://user-images.githubusercontent.com/43215253/78648861-84761900-78b4-11ea-9805-e479a3edc8c8.png">

By the way, I opened a separate PR in #40834 proposing switching the buttons of the Support Article Dialog for the Free Photo Library card. I figured it looks odd when this card proposes them in this order. :)


<img width="240" alt="Screenshot 2020-04-07 at 09 46 14" src="https://user-images.githubusercontent.com/43215253/78648944-a7083200-78b4-11ea-94ed-6f05a0f00517.png">

kwight commented:
> Just to clarify, should the list still be those four things or should it be "Adjusting settings for blocks" and "Customizing posts and pages with blocks", in which case the whole video could be played?

Let's go with the second case, rather than relying on chopping up the video experience.

> By the way, I opened a separate PR in #40834 proposing switching the buttons of the Support Article Dialog for the Free Photo Library card.

Ha, I was just wondering that myself today too. We'll let design give final guidance in that one. 

You can put this component under `client/my-sites/customer-home/cards/primary` for now, and maybe just suggest how to trigger it for viewing in the Testing Instructions. The location layout may be changing soon, but this will keep us all moving forward. Thanks again! <3

Aurorum commented:
Thanks @kwight! Out of curiosity though, wouldn't they be best suited for `client/my-sites/customer-home/cards/education`? I have no strong preference either way though. :)

kwight commented:
@Aurorum I was just going with the idea that (from what I could tell above, for now) it would be displayed in the primary location, but yep, either is fine 👍

gwwar commented on client/my-sites/customer-home/main.jsx:
```diff
@@ -169,4 +169,4 @@ const mergeProps = ( stateProps, dispatchProps, ownProps ) => {
 const connectHome = connect( mapStateToProps, mapDispatchToProps, mergeProps );
 
-export default flowRight( connectHome, withTrackingTool( 'HotJar' ) )( Home );
+export default flowRight( connectHome, withTrackingTool( 'HotJar' ) )( Home );
```
Pretty minor but the linter expects an endline at the end of files, which is why the linter failed.

gwwar commented:
❤️ Thanks @Aurorum for looking at this one again. The following location would work for now! `client/my-sites/customer-home/cards/education`

In terms of ordering for next PRs I'm thinking we might have:
- Add in this new tutorial card using the same patterns as the last one. (Can render with testing on the client but not visible yet in prod).
- Since we now have more than one tutorial, and more coming, we'll want a follow up PR to group videos in either a combined section or card (we'll have some sort of pager). @danhauk is thinking through options.
- Finally wire in the new tutorial grouping into the Home layout API

Aurorum commented:
Okay, I think this should be ready for review now. I've updated the testing instructions to provide a bit more detail after the changes! :)

gwwar commented on client/my-sites/customer-home/cards/education/mastering-gutenberg/index.jsx:
```diff
@@ -0,0 +1,122 @@
+
+export default connect( state => {
+	return {
+		cannotUseGutenberg: isClassicEditorForced( state, getSelectedSiteId( state ) ),
```
Good thinking here! 👍 We'll probably move this check over to the layout API but this is a great reminder ✨

Aurorum replied:
Should I remove it here now? Just checking as it seems other cards have display logic within them too. :)

https://github.com/Automattic/wp-calypso/blob/1d38c53dba02a5930b39fac2ab3c23f7ba52d6a1/client/my-sites/customer-home/cards/features/stats/index.jsx#L37-L39

https://github.com/Automattic/wp-calypso/blob/1d38c53dba02a5930b39fac2ab3c23f7ba52d6a1/client/my-sites/customer-home/cards/features/grow-earn/index.jsx#L32-L34

mmtr replied:
We can follow up later once they are moved to the API, no need to remove them for now.

Note that `isClassicEditorForced` will return false when the selected editor is the classic one. So I wonder if we should show the card only when the selected editor is Gutenberg: `cannotUseGutenberg: getSelectedEditor( state, siteId ) === 'classic'`;

mmtr replied:
> We can follow up later once they are moved to the API, no need to remove them for now.

Actually, since the card won't be displayed until we update the API, let's remove the checks now. We'll make sure the card won't be returned by the API is Gutenberg is not the selected editor.

Aurorum replied:
> So I wonder if we should show the card only when the selected editor is Gutenberg: cannotUseGutenberg: getSelectedEditor( state, siteId ) === 'classic'

There was some discussion about this [here](https://github.com/Automattic/wp-calypso/pull/39617#issuecomment-591086780):

> While I agree that it would make sense to hide the card if a person explicitly doesn't want to use Gutenberg, the community keeps adding new functionality and fixing things that didn't work well in a previous release. A Dismiss option works best for serving all of those intentions.

Now that it's just an ordinary card and not a dismissable one, I'm not sure that still applies. 

I've removed the checks anyway. :)

gwwar commented on client/my-sites/customer-home/cards/education/mastering-gutenberg/index.jsx:
```diff
@@ -0,0 +1,122 @@
+						title={ translate( 'Video demonstration' ) }
+						width="560"
+						height="315"
+						src={ `https://www.youtube.com/embed/${ videoId }?rel=0&autoplay=1` }
```
Looking back at the free image library card it looks like we ended up updating the support page (adding the video to the top) and reusing the Support Article Dialog. How do folks feel if follow that pattern and update support articles with the new videos. I'm also okay with the showing the videos directly if we update the modal dialog size.

| Free Images | Mastering the Block Editor |
| ------------- | ------------- |
| <img width="1568" alt="Screen Shot 2020-04-07 at 4 10 32 PM" src="https://user-images.githubusercontent.com/1270189/78727883-c1f59900-78ea-11ea-85a7-3e14c8348b6f.png"> | <img width="1554" alt="Screen Shot 2020-04-07 at 4 10 43 PM" src="https://user-images.githubusercontent.com/1270189/78727962-f36e6480-78ea-11ea-95e9-33edda27b8e3.png"> |

If folks are okay with the suggestion @mmtr or @kwight let's update https://wordpress.com/support/wordpress-editor/ to add the first video at the top.

@arunsathiya do you happen to know offhand of a good support page match for [customizing pages and posts](https://www.youtube.com/watch?v=BkQlTuJ4BH8)?

Aurorum replied:
I'm finding it a challenge to make the video responsive, although I've made it a bit larger. It's probably better to embed it in the support document, but I'm not sure there's a specific document to match the contents of the video like there is for the Free Photo Library.

mmtr replied:
[This commit](https://github.com/Automattic/wp-calypso/pull/40080/commits/7eea97e01b1d0d83b8349f5e7df7cce698a1c89e) might help you. We used that back when the free photo library card only displayed a youtube video.

mmtr replied:
But yeah, in general I prefer to follow the pattern of showing a support article dialog that contains the video as we are doing for the new checklist task for updating site menus (https://github.com/Automattic/wp-calypso/pull/40844). I'll update the support articles with the videos, so we can use them here.

mmtr replied:
> If folks are okay with the suggestion @mmtr or @kwight let's update https://wordpress.com/support/wordpress-editor/ to add the first video at the top.

Not sure if that's the best article for https://www.youtube.com/watch?v=oQNap9YTrWw. The video already starts with a block inserted but seems that https://wordpress.com/support/wordpress-editor/ tries to provide guidance for more basic stuff (such as inserting a block). 

Actually, I think https://www.youtube.com/watch?v=BkQlTuJ4BH8 is a better fit for https://wordpress.com/support/wordpress-editor/ since it shows how to add blocks, arrange and move them around which matches the content of the support article.

@arunsathiya would you agree?

arunsathiya replied:
>@arunsathiya would you agree?

Yup, I agree, @mmtr! 

>@arunsathiya do you happen to know offhand of a good support page match for customizing pages and posts?

https://wordpress.com/support/wordpress-editor/ comes closest. It has also been a while since I did WPcom support, and this page has changed a lot since my last visit. :) I also noticed that this YouTube has a mention of how to add images from the media library, but it's not covered within this support document. https://wordpress.com/support/wordpress-editor/blocks/image-block/ does that instead. It's only a small part of the video though, so, I guess we can continue to consider https://wordpress.com/support/wordpress-editor/ as the best match.

mmtr replied:
Thanks @arunsathiya! Actually, after taking a deeper look at https://wordpress.com/support/wordpress-editor/ and the videos, I think both videos could be included in the same article.

https://www.youtube.com/watch?v=BkQlTuJ4BH8 covers [Adding a Block](https://wordpress.com/support/wordpress-editor/#adding-a-block), [Moving Blocks](https://wordpress.com/support/wordpress-editor/#moving-blocks), [Block Navigation](https://wordpress.com/support/wordpress-editor/#block-navigation) and [Removing a Block](https://wordpress.com/support/wordpress-editor/#removing-a-block) while https://www.youtube.com/watch?v=oQNap9YTrWw covers [Configuring a Block](https://wordpress.com/support/wordpress-editor/#configuring-a-block).

Unfortunately, the inline support dialog cannot point to a specific part of an article, so the options we have are:
- Add both videos to https://wordpress.com/support/wordpress-editor. The card should include then one single link to that article.
- Keep showing videos directly without loading any support article.
- Add only https://www.youtube.com/watch?v=BkQlTuJ4BH8 to https://wordpress.com/support/wordpress-editor. The 1st link of the card (_Adjusting setting of blocks_) should open directly. The 2nd link (_Customizing pages and posts with blocks_) will open the support article.

@Automattic/dotcom-manage-design @Automattic/manage do you have any preference?

gwwar replied:
> Unfortunately, the inline support dialog cannot point to a specific part of an article, so the options we have are:

Hmm, can we not load to an anchor point? Seems like something we can update in a separate PR

gwwar replied:
> Unfortunately, the inline support dialog cannot point to a specific part of an article, so the options we have are:

I've added this in https://github.com/Automattic/wp-calypso/pull/40937

gwwar replied:
I also added the videos in https://wordpress.com/support/wordpress-editor/#blocks and https://wordpress.com/support/wordpress-editor/#configuring-a-block

gwwar commented on client/my-sites/customer-home/cards/education/mastering-gutenberg/index.jsx:
```diff
@@ -0,0 +1,122 @@
+/**
+ * Internal dependencies
+ */
+import DismissibleCard from 'blocks/dismissible-card';
```
We can probably drop the DismissableCard option, since I think we'll end up favoring letting folks cycle through the educational cards, right @danhauk ?

Aurorum replied:
Switched from a `DismissibleCard ` to an ordinary `Card` 👍

gwwar commented:
Nice work @Aurorum ! This is looking great so far ✨

mmtr commented:
Just a heads up that I just merged #40937 which adds to the support article dialog the ability to scroll to an anchor, which effectively unblocks this PR.

Aurorum commented:
Thank you! I've updated the PR to use the support article dialog - much less code needed too! :)

Aurorum commented:
Actually, just want to check if "adjusting settings of blocks" and "customizing posts and pages with blocks" should still be the links. Although they apply better to the video, the previous links seem better suited to the content of the support doc.

mmtr commented:
> Actually, just want to check if "adjusting settings of blocks" and "customizing posts and pages with blocks" should still be the links. Although they apply better to the video, the previous links seem better suited to the content of the support doc.

Since the video is the primary content that will be shown in the dialog, I think the copy of the links still match what the user will see in the dialog.

mmtr commented on client/my-sites/customer-home/cards/education/mastering-gutenberg/index.jsx:
```diff
@@ -0,0 +1,65 @@
+	return {
+		cannotUseGutenberg: isClassicEditorForced( state, getSelectedSiteId( state ) ),
+	};
+} )( localize( MasteringGutenberg ) );
```
We also have the `useTranslate` hook, which simplifies a bit the code on functional components like this.

mmtr commented on client/my-sites/customer-home/cards/education/mastering-gutenberg/index.jsx:
```diff
@@ -0,0 +1,65 @@
+				<CardHeading>{ translate( 'Master the Block Editor' ) }</CardHeading>
+				<p className="mastering-gutenberg__text customer-home__card-subheader">
+					{ translate(
+						'Learn how to create stunning post and page layouts ' + 'through our video guides.'
```
Seems we don't need to concat the 2 parts here.
```suggestion
						'Learn how to create stunning post and page layouts through our video guides.'
```

Aurorum replied:
Hm, not sure why I did that to begin with. Fixed. :)

Aurorum commented on client/my-sites/customer-home/cards/education/mastering-gutenberg/index.jsx:
```diff
@@ -0,0 +1,65 @@
+				</p>
+				<InlineSupportLink
+					supportPostId={ 147594 }
+					supportLink={ localizeUrl( 'https://wordpress.com/support/wordpress-editor/#blocks' ) }
```
Fixed the links (oops) and I switched the order. :) 

<img width="744" alt="Screenshot 2020-04-09 at 16 59 11" src="https://user-images.githubusercontent.com/43215253/78915184-7296a080-7a83-11ea-850d-e794b1c35b7e.png">

Aurorum commented:
The Free Photo Library illustration causes the dialog to appear. I wonder if it'd be a bit confusing that the Mastering Gutenberg doesn't do the same? 

<img width="316" alt="Screenshot 2020-04-09 at 17 02 57" src="https://user-images.githubusercontent.com/43215253/78915525-f8b2e700-7a83-11ea-9ceb-3030fe6674fa.png">

gwwar commented:
> The Free Photo Library illustration causes the dialog to appear. I wonder if it'd be a bit confusing that the Mastering Gutenberg doesn't do the same?

I think that's an artifact of us starting with an icon that had a play button originally. I think not triggering the dialog is okay here. We can tidy up the other card behavior in another PR.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/3203786

Thank you @Aurorum for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/3203786).
